File: DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/Controls/InfoViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Input
Imports DevExpress.Xpf
Imports DevExpress.Mvvm
Imports DevExpress.Utils

Namespace ProductsDemo
    Public Class InfoViewModel
        Inherits BindableBase


        Private aboutInfo_Renamed As AboutInfo

        Private showHelpCommand_Renamed As ICommand

        Private showGettingStartedCommand_Renamed As ICommand

        Private showContactUsCommand_Renamed As ICommand

        Public ReadOnly Property AboutInfo() As AboutInfo
            Get
                If aboutInfo_Renamed Is Nothing Then
                    aboutInfo_Renamed = New AboutInfo()
                    Dim data As DevExpress.Internal.UserData = DevExpress.Utils.About.Utility.GetInfoEx()
                    aboutInfo_Renamed.LicenseState = If(data.IsExpired, LicenseState.TrialExpired, If(data.IsTrial, LicenseState.Trial, LicenseState.Licensed))
                    aboutInfo_Renamed.ProductName = "WPF Controls"
                    aboutInfo_Renamed.ProductPlatform = "Build Your Own Office"
                    aboutInfo_Renamed.RegistrationCode = WpfSerialNumberProvider.GetSerial()
                    aboutInfo_Renamed.Version = AssemblyInfo.Version
                End If
                Return aboutInfo_Renamed
            End Get
        End Property
        Public Sub ShowHelp()
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("http://help.devexpress.com")
        End Sub
        Public Sub ShowGettingStarted()
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("http://devexpress.com/support/")
        End Sub
        Public Sub ShowContactUs()
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("http://devexpress.com/Support/Center")
        End Sub
        Public ReadOnly Property ShowHelpCommand() As ICommand
            Get
                If showHelpCommand_Renamed Is Nothing Then
                    showHelpCommand_Renamed = New DelegateCommand(AddressOf ShowHelp, False)
                End If
                Return showHelpCommand_Renamed
            End Get
        End Property
        Public ReadOnly Property ShowGettingStartedCommand() As ICommand
            Get
                If showGettingStartedCommand_Renamed Is Nothing Then
                    showGettingStartedCommand_Renamed = New DelegateCommand(AddressOf ShowGettingStarted, False)
                End If
                Return showGettingStartedCommand_Renamed
            End Get
        End Property
        Public ReadOnly Property ShowContactUsCommand() As ICommand
            Get
                If showContactUsCommand_Renamed Is Nothing Then
                    showContactUsCommand_Renamed = New DelegateCommand(AddressOf ShowContactUs, False)
                End If
                Return showContactUsCommand_Renamed
            End Get
        End Property
    End Class
End Namespace

File: DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/Controls/InfoViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Input;
using DevExpress.Xpf;
using DevExpress.Mvvm;
using DevExpress.Utils;

namespace ProductsDemo {
    public class InfoViewModel : BindableBase {
        AboutInfo aboutInfo;
        ICommand showHelpCommand;
        ICommand showGettingStartedCommand;
        ICommand showContactUsCommand;

        public AboutInfo AboutInfo {
            get {
                if(aboutInfo == null) {
                    aboutInfo = new AboutInfo();
                    DevExpress.Internal.UserData data = DevExpress.Utils.About.Utility.GetInfoEx();                    
                    aboutInfo.LicenseState = data.IsExpired ? LicenseState.TrialExpired : data.IsTrial ? LicenseState.Trial : LicenseState.Licensed;
                    aboutInfo.ProductName = "WPF Controls";
                    aboutInfo.ProductPlatform = "Build Your Own Office";
                    aboutInfo.RegistrationCode = WpfSerialNumberProvider.GetSerial();
                    aboutInfo.Version = AssemblyInfo.Version;
                }
                return aboutInfo;
            }
        }
        public void ShowHelp() {
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("http://help.devexpress.com");
        }
        public void ShowGettingStarted() {
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("http://devexpress.com/support/");
        }
        public void ShowContactUs() {
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("http://devexpress.com/Support/Center");
        }
        public ICommand ShowHelpCommand {
            get {
                if(showHelpCommand == null)
                    showHelpCommand = new DelegateCommand(ShowHelp, false);
                return showHelpCommand;
            }
        }
        public ICommand ShowGettingStartedCommand {
            get {
                if(showGettingStartedCommand == null)
                    showGettingStartedCommand = new DelegateCommand(ShowGettingStarted, false);
                return showGettingStartedCommand;
            }
        }
        public ICommand ShowContactUsCommand {
            get {
                if(showContactUsCommand == null)
                    showContactUsCommand = new DelegateCommand(ShowContactUs, false);
                return showContactUsCommand;
            }
        }
    }
}
